operator/v1/types.go

```diff
@@ -258,15 +258,21 @@ type StaticPodOperatorStatus struct {
 
 // NodeStatus provides information about the current state of a particular node managed by this operator.
 // +kubebuilder:validation:XValidation:rule="has(self.currentRevision) || !has(oldSelf.currentRevision)",message="cannot be unset once set",fieldPath=".currentRevision"
+// +kubebuilder:validation:XValidation:rule="oldSelf.hasValue() || (has(self.currentRevision) ? self.currentRevision == 0 : true) ",message="when specified on creation of a nodeStatus, currentRevision must be set to 0 on creation",optionalOldSelf=true
```

Comment:
Not saying this is better, but I think you could also write this as
```suggestion
// +kubebuilder:validation:XValidation:rule="oldSelf.hasValue() || self.?currentRevision.orValue(0) == 0) ",message="when specified on creation of a nodeStatus, currentRevision must be set to 0 on creation",optionalOldSelf=true
```

Reply:
Perfectly valid, was just giving you a second option in case you felt this way might have been easier, I'm happy with either